metabase/metabase
# feat(sdk): customize dashboard, dashcard and question background color

Closes https://github.com/metabase/metabase/issues/42678

### Description

Exposes the following theme colors to the user for customizing their static dashboard and static question colors.

- `background-dashboard`: background color of dashboards, default to `background` (`bg-white`)
- `background-dashboard-card`: background color of dashboard cards, default to `background` (`bg-white`)
- `background-question`: background color of questions, default to unset.

### How to verify

- Go to the `test-dashcard-background` branch in the demo app
- Go to `http://localhost:3004/dev`, you should see a static dashboard and some static questions in the Kitchen Sink.
- Change the above colors in `theme.colors`, and the color should change accordingly.

### Demo

Setting `colors.text-*` and `colors.background` results in a transparent-ish theme.

<img src="https://github.com/metabase/metabase/assets/4714175/a53c5374-c409-486a-a704-504387766730" width="400">

Adding `theme.components.dashboard.card.backgroundColor` adds background to the card.

<img src="https://github.com/metabase/metabase/assets/4714175/65298286-978a-45ae-84f3-2aee6a09eab9" width="400">

Adding `theme.components.question.backgroundColor` sets the question background color, and `theme.components.dashboard.backgroundColor` sets the background color to the dashboard component.

<img src="https://github.com/metabase/metabase/assets/4714175/63b41044-7dfa-4725-bd0b-cce9dee3fa4c" width="400">

### Caveats

- There is no stories/visual tests and unit tests that renders static dashboard yet, so I wasn't able to write tests for this. @oisincoveney any recommendations on rendering the static dashboard in a Storybook environment?
